Documentation/development-environment.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Multi-Node Test Environment
+weight: 91
+indent: true
+---
+
+# Rook development workflow
+
+## Setup expectation
+
+There are a bunch of pre-requisites to be able to deploy the following environment. Such as:
+
+* A Linux workstation (CentOS or Fedora)
+* KVM/QEMU installation
+
+For other Linux distribution, there is no guarantee the following will work.
+However adapting commands (apt/yum/dnf) could just work.
+
+## Prerequisites installation
+
+Execute `tests/scripts/multi-node/redhat-system-prerequisites.sh`.
+
+## Deploy Kubernetes with Kubespray
+
+In order to successfully deploy Kubernetes with Kubespray, you must have this code: https://github.com/kubernetes-incubator/kubespray/pull/2153 and https://github.com/kubernetes-incubator/kubespray/pull/2271.
+
+Edit `inventory/group_vars/k8s-cluster.yml` with:
+
+```bash
+docker_options: "--insecure-registry=172.17.8.1:5000 --insecure-registry={{ kube_service_addresses }} --graph={{ docker_daemon_graph }}  {{ docker_log_opts }}"
+```
+
+FYI: `172.17.8.1` is the libvirt bridge IP, so it's reachable from all your virtual machines.
+If you use k8s-01 as a registry, you should change it, either by a subnet of the IP of that machine in the `172.17.8.0/24` subnet.
+
+Create Vagrant's variable directory:
+
+```bash
+mkdir vagrant/
+```
+
+Create `vagrant/config.rb` and use the content from : `tests/scripts/multi-node/config.rb`. You can adapt it at will.
+Feel free to adapt `num_instances`.
+
+Deploy!
+
+```bash
+vagrant up --no-provision ; vagrant provision
```

Comment:
why do we need the no-provision and the provision steps separately?

Reply:
just a vagrant thing, sometimes the provisioning sequence fails and/or doesn't start properly or wait for all the vms to be available. This is not mandatory, more a safety measure.